SU2_CFD/src/numerics/CFEAMeshElasticity.cpp

```diff
@@ -0,0 +1,122 @@
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../include/numerics/CFEAMeshElasticity.hpp"
+#include <limits>
+
+CFEAMeshElasticity::CFEAMeshElasticity(unsigned short val_nDim, unsigned short val_nVar, unsigned long val_nElem, CConfig *config) : CFEALinearElasticity() {
+
+  DV_Val         = NULL;
+  FAux_Dead_Load = NULL;
+  Rho_s_i        = NULL;
+  Rho_s_DL_i     = NULL;
+  Nu_i           = NULL;
+
+  nDim = val_nDim;
+  nVar = val_nVar;
+
+  unsigned long iVar;
+
+  E = config->GetDeform_ElasticityMod();
+  Nu = config->GetDeform_PoissonRatio();
+  Compute_Lame_Parameters();
+
+  switch (config->GetDeform_Stiffness_Type()) {
+  case INVERSE_VOLUME:
+  case SOLID_WALL_DISTANCE:
+    element_based = true;
+    Nu = config->GetDeform_Coeff();
+    break;
+  case CONSTANT_STIFFNESS:
+    element_based = false;
+    break;
+  }
+
+  E_i  = NULL;
+  if (element_based){
+    E_i         = new su2double[val_nElem];
+    for (iVar = 0; iVar < val_nElem; iVar++){
+      E_i[iVar] = E;
+    }
+  }
+
+  KAux_ab = new su2double* [nDim];
+  for (iVar = 0; iVar < nDim; iVar++) {
+    KAux_ab[iVar] = new su2double[nDim];
+  }
+
+  if (nDim == 2) {
+    Ba_Mat = new su2double* [3];
+    Bb_Mat = new su2double* [3];
+    D_Mat  = new su2double* [3];
+    Ni_Vec  = new su2double [4];            /*--- As of now, 4 is the maximum number of nodes for 2D problems ---*/
+    GradNi_Ref_Mat = new su2double* [4];
+    GradNi_Curr_Mat = new su2double* [4];
+    for (iVar = 0; iVar < 3; iVar++) {
+      Ba_Mat[iVar]      = new su2double[nDim];
+      Bb_Mat[iVar]      = new su2double[nDim];
+      D_Mat[iVar]       = new su2double[3];
+    }
+    for (iVar = 0; iVar < 4; iVar++) {
+      GradNi_Ref_Mat[iVar]   = new su2double[nDim];
+      GradNi_Curr_Mat[iVar]   = new su2double[nDim];
+    }
+  }
+  else if (nDim == 3) {
+    Ba_Mat = new su2double* [6];
+    Bb_Mat = new su2double* [6];
+    D_Mat  = new su2double* [6];
+    Ni_Vec  = new su2double [8];           /*--- As of now, 8 is the maximum number of nodes for 3D problems ---*/
+    GradNi_Ref_Mat = new su2double* [8];
+    GradNi_Curr_Mat = new su2double* [8];
+    for (iVar = 0; iVar < 6; iVar++) {
+      Ba_Mat[iVar]      = new su2double[nDim];
+      Bb_Mat[iVar]      = new su2double[nDim];
+      D_Mat[iVar]        = new su2double[6];
+    }
+    for (iVar = 0; iVar < 8; iVar++) {
+      GradNi_Ref_Mat[iVar]   = new su2double[nDim];
+      GradNi_Curr_Mat[iVar]   = new su2double[nDim];
+    }
+  }
+
+}
+
+CFEAMeshElasticity::~CFEAMeshElasticity(void) {
+
```

Comment:
Make sure that all of the things allocated in the constructor is deleted properly in the destructor

Reply:
I believe that everything is already being deleted in the destructor of the base class CFEAElasticity::~CFEAElasticity() (see numerics_direct_elasticity.cpp)